subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/ResolvableConfigurationBuilder.java

```diff
@@ -0,0 +1,94 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.plugins.jvm;
+
+import org.gradle.api.Action;
+import org.gradle.api.Incubating;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.provider.Provider;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * An incoming configuration builder. Such a configuration is meant
+ * to be resolved.
+ *
+ * @since 6.7
+ */
+@Incubating
+public interface ResolvableConfigurationBuilder {
+
+    /**
+     * Sets the description of the resolvable configuration this builder creates
+     * @param description the description
+     */
+    ResolvableConfigurationBuilder withDescription(String description);
+
+    /**
+     * Also create the dependency bucket of the provided name and
+     * make the resolvable configuration extend from it.
+     * @param name the name of the bucket of dependencies
+     */
+    ResolvableConfigurationBuilder usingDependencyBucket(String name);
+
+    /**
+     * Also create the dependency bucket of the provided name and
+     * make the resolvable configuration extend from it.
+     * @param name the name of the bucket of dependencies
+     * @param description a description for this dependency bucket
+     */
+    ResolvableConfigurationBuilder usingDependencyBucket(String name, String description);
+
+    /**
+     * Configures the resolution for runtime of java libraries
+     */
+    ResolvableConfigurationBuilder requiresJavaLibrariesRuntime();
+
+    /**
+     * Configures the resolution for API of java libraries
+     */
+    ResolvableConfigurationBuilder requiresJavaLibrariesAPI();
+
+    /**
+     * Adds configurations the resolvable configuration should extend from.
+     * Those configurations should be typically buckets of dependencies
+     * @param parentConfigurations the parent configurations
+     */
+    ResolvableConfigurationBuilder extendsFrom(Configuration... parentConfigurations);
+
+    /**
+     * Adds configurations the resolvable configuration should extend from.
+     * Those configurations should be typically buckets of dependencies
+     * @param parentConfigurations the parent configurations
+     */
+    ResolvableConfigurationBuilder extendsFrom(List<Provider<Configuration>> parentConfigurations);
+
+    /**
+     * Adds this configuration as a parent configuration of the resolvable configuration
+     * @param configuration the parent configuration
+     */
+    default ResolvableConfigurationBuilder extendsFrom(Provider<Configuration> configuration) {
+        return extendsFrom(Collections.singletonList(configuration));
+    }
+
+    /**
+     * Allows refining the attributes of this configuration in case the defaults are not
+     * sufficient. The refiner will be called after the default attributes are set.
```

Comment:
Is there anything except 'usage==runtime'? I find this a bit confusing:

> The refiner will be called after the default attributes are set.

As it is unclear what "the default attributes" are and where they come from.